### PR TITLE
[WFLY-16412] Failed deployment leaves classloader leaks through undertow references

### PR DESCRIPTION
**Pull request by aogburn**
https://issues.redhat.com/browse/WFLY-16412

### CODE REVIEW

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java [hunk 1]
```diff
@@ -427,6 +427,12 @@ public synchronized void start(final StartContext startContext) throws StartExce
 
     @Override
     public synchronized void stop(final StopContext stopContext) {
+        // Remove the server activity
+        final ServerActivity activity = serverActivity.get();
+        if(activity != null) {
+            suspendController.get().unRegisterActivity(activity);
+        }
+        UndertowContainerProvider.removeContainer(module.getClassLoader());
```
Comment (aogburn) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java:435]: I think we'd want to actually call io.undertow.websockets.jsr.SecurityActions.removeContainer instead but this is protected:
https://github.com/undertow-io/undertow/blob/master/websockets-jsr/src/main/java/io/undertow/websockets/jsr/SecurityActions.java#L40

So do we change that from protected?  Or add the security manager handling to UndertowDeploymentInfoService?

Comment (bstansberry) [reply undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java:435]: It shouldn't matter. An MSC thread calling stop shouldn't have anything but container classes on the call stack.

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java [hunk 2]
```diff
@@ -949,19 +954,6 @@ public void resume() {
...
-
-                    @Override
-                    public void contextDestroyed(ServletContextEvent sce) {
-                        final ServerActivity activity = serverActivity.get();
-                        if(activity != null) {
-                            suspendController.get().unRegisterActivity(activity);
-                        }
-                    }
-                };
-                d.addListener(new ListenerInfo(sl.getClass(), new ImmediateInstanceFactory<EventListener>(sl)));
```
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java:964]: I don't know enough about this code to know if removing this is ok, but if there are concerns from those who do I think you can achieve your objective by leaving this here and replacing serverActivity.get() both here and in the stop method with serverActivity.getAndSet(null).

Comment (bmaxwell) [reply undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java:964]: @aogburn / @fl4via  can you check this one, it looks like the fix has been stalled for several months

Comment (fl4via) [reply undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java:964]: yes, it is okay to remove this block.

**Review by fl4via — CHANGES_REQUESTED** [review]
it is just pending the security manager handling

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java (hunk 1 above)
Comment (aogburn) [reply undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java]: Thanks @fl4via, added it to UndertowDeploymentInfoService

**Review by fl4via — APPROVED** [review]

**Pull request merged**